src/x/instrument/extended.go

```diff
@@ -126,18 +125,102 @@ func StartReportingExtendedMetrics(
 	return reporter, nil
 }
 
+// cumulativeCounter emits a cumulative memory stat counter as a metric counter.
+// the counter tracks the last value, so it can emit deltas to the underlying counter metric.
+type cumulativeCounter struct {
+	counter tally.Counter
+	last    uint64
+}
+
+func newCumulativeCounter(scope tally.Scope, name string) cumulativeCounter {
+	return cumulativeCounter{
+		counter: scope.Counter(name),
+	}
+}
+
+// update the cumulative counter and returns the previous last value.
+func (c *cumulativeCounter) update(new uint64) uint64 {
+	prevLast := c.last
+	c.counter.Inc(int64(new - c.last))
```

Comment:
if all of these were `int64` we could avoid the cast?

Reply:
yea. we have to cast somewhere since go MemStats exposed `uint64`. seems best to cast the delta, which should be much smaller than the cumulative counter.